### ISSUE TITLE
logits_all issue?

### ISSUE THREAD

**Issue by sergeigofman** [zhougr1993/DeepInterestNetwork#7]
Hi,
My understanding is that logits_all symbol is used for real inference scenario, in which interactions of a specific user with all possible items are checked.
If this is true, then the fact that u_emb represents attention of a user history with unrelated item i, and then gets broadcasted along all items, seems wrong to me. Instead, I would expect that each item would be attended vs. user's history.
Appreciate your comments!

**Comment by zhougr1993** [comment]
Thanks a lot. My fault! I am so sorry for the low-quality code.
Actually the logits_all is irrelevant. AUC is calculated on self.i and self.j. 
The calculation of logits_all is wrong. I will delete it.

**Comment by zhougr1993** [comment]
This issue has been fixed by https://github.com/zhougr1993/DeepInterestNetwork/commit/58dcc8d816f625402831071bad9923f5107a77d2.

**Comment by sergeigofman** [comment]
Thanks for the fix. I am in the process of understanding the code that was added. One thing I noted, though, is that the default activation for the predictions (d_layer_1_sub, d_layer_2_sub) is dice while for the training path it is sigmoid. I believe this should be made consistent, probably making sigmoid a default and dice optional along the lines of the code that was there already.
